### ISSUE TITLE
Unclear log output on incorrect permissions of subdirectory

### ISSUE THREAD

**Issue by rule88** [elastic/beats#4621]
When one uses a subdirectory (e.g. conf.d) for its prospectors, no clear error message will be generated when (group) permissions are incorrect. It would save one a lot of time when the error message does says more about why it isn't loading the sub-configuration files.

For confirmed bugs, please report:
- Filebeat-5.4.3 (upgraded directly from 5.1.1) 
- RHEL 7.3
- Steps to Reproduce: 

```
[root@host filebeat]# ls -hal conf.d/
total 8.0K
drwxr-xr-x 2 root root 68 Jul 6 10:28 .
drwxr-xr-x. 3 root root 40 Jul 6 10:28 ..
-rw-rw-r-- 1 root root 198 Jul 6 10:26 filebeat-prospector.yml
-rw-rw-r-- 1 root root 194 Jul 6 10:26 logstash-prospector.yml
[root@host filebeat]# /usr/bin/filebeat.sh -v -c /etc/filebeat/filebeat.yml -configtest -d "*"
Exiting: No prospectors defined. What files do you want me to watch?
[root@host filebeat]# chmod g-w conf.d/*
[root@host filebeat]# /usr/bin/filebeat.sh -v -c /etc/filebeat/filebeat.yml -configtest -d "*"
Config OK
```

From filebeat the following config parameter is used to load from subdir:

`filebeat.config_dir: /etc/filebeat/conf.d`

ref: https://discuss.elastic.co/t/unclear-output-on-incorrect-permissions-of-subdirectory/92053

**Comment by exekias** [comment]
As a note, metricbeat may also be affected by this with `modules` reload

**Comment by andrewkroh** [comment]
Do you get any more info if you enable logging to stderr (`-e`) when running config test?

**Comment by rule88** [comment]
Same non explanatory output:

`[root@host filebeat]# /usr/bin/filebeat.sh -e -v -c /etc/filebeat/filebeat.yml -configtest -d "*" 
2017/07/06 18:52:58.745615 beat.go:285: INFO Home path: [/usr/share/filebeat] Config path: [/etc/filebeat] Data path: [/var/lib/filebeat] Logs path: [/var/log/filebeat]
2017/07/06 18:52:58.745680 beat.go:186: INFO Setup Beat: filebeat; Version: 5.4.3
2017/07/06 18:52:58.745817 metrics.go:23: INFO Metrics logging every 30s
2017/07/06 18:52:58.745890 logstash.go:90: INFO Max Retries set to: 3
2017/07/06 18:52:58.746054 outputs.go:108: INFO Activated logstash as output plugin.
2017/07/06 18:52:58.746238 publish.go:295: INFO Publisher name: webserver01.int.rule88.nl
2017/07/06 18:52:58.746588 async.go:63: INFO Flush Interval set to: 1s
2017/07/06 18:52:58.746619 async.go:64: INFO Max Bulk Size set to: 2048
2017/07/06 18:52:58.747001 config.go:114: INFO Additional config files are fetched from: /etc/filebeat/conf.d
2017/07/06 18:52:58.747101 config.go:87: INFO Additional configs loaded from: /etc/filebeat/conf.d/prospector.yml
2017/07/06 18:52:58.747164 beat.go:339: CRIT Exiting: No prospectors defined. What files do you want me to watch?
Exiting: No prospectors defined. What files do you want me to watch?`

**Comment by rule88** [comment]
and with permissions corrected on conf.d files (mode 644):

```
[root@host filebeat]# /usr/bin/filebeat.sh -e -v -c /etc/filebeat/filebeat.yml -configtest -d "*" 
2017/07/06 18:55:04.145121 beat.go:285: INFO Home path: [/usr/share/filebeat] Config path: [/etc/filebeat] Data path: [/var/lib/filebeat] Logs path: [/var/log/filebeat]
2017/07/06 18:55:04.145206 beat.go:186: INFO Setup Beat: filebeat; Version: 5.4.3
2017/07/06 18:55:04.145417 logstash.go:90: INFO Max Retries set to: 3
2017/07/06 18:55:04.145562 outputs.go:108: INFO Activated logstash as output plugin.
2017/07/06 18:55:04.145644 metrics.go:23: INFO Metrics logging every 30s
2017/07/06 18:55:04.145729 publish.go:295: INFO Publisher name: webserver01.int.rule88.nl
2017/07/06 18:55:04.146049 async.go:63: INFO Flush Interval set to: 1s
2017/07/06 18:55:04.146077 async.go:64: INFO Max Bulk Size set to: 2048
2017/07/06 18:55:04.146481 config.go:114: INFO Additional config files are fetched from: /etc/filebeat/conf.d
2017/07/06 18:55:04.146568 config.go:87: INFO Additional configs loaded from: /etc/filebeat/conf.d/prospector.yml
Config OK
```

**Comment by michalpristas** [comment]
Fixed by #3573
